### DISCUSSION TITLE
Script no?

### DISCUSSION THREAD

**Discussion by Dregi56** [indomus/forum#323] [HUB: Home Assistant]
Ho notato che se in Alexa lancio la ricerca dispositivi, gli script non vengono mostrati, anche se questi sono settati come esposti nel filter include_entities di smart home in configuration.yaml.
Ora la domanda è in che modo posso "chiamare su Alexa" l'esecuzione di uno script?

**Comment by snakuzzo** [comment]
Ciao.
Script, scene ed automazioni non li trovi come dispositivi.
Per vederli devi creare una routine e lì poi potrai selezionare script, scene ed automazioni che hai esposto verso il tuo assistente vocale.

P.S. Questa cosa vale anche per Google Assistant

**Reply by KC-inDomus** [reply]
Ti sbagli Diego, gli scenari HA su Alexa vengono visti direttamente come scenari.

**Reply by snakuzzo** [reply]
ok...quindi vale per tutto tranne le scene ?

**Comment by indomus** [comment]
Gli script vengono visti come scenari, stai sbagliando qualcosa. https://indomus.it/progetti/eseguire-sequenze-logiche-sulla-domotica-home-assistant-tramite-comandi-vocali-alexa/

**Comment by Dregi56** [comment]
Ok. Perfetto. Grazie. Si sono visibili tra le scene, comunque si riesce ad eseguirli da Alexa solo creando una routine.

**Reply by KC-inDomus** [reply]
veramente ti basta dire "Alexa accendi" affiancato al nome dello script.

**Reply by Dregi56** [reply]
Si, certo, ho poi visto che questa strada è percorribile, anche se non è il massimo. Sarebbe opportuno avere la possibilità di configurare a piacere le "utterances" d'innesco.

**Reply by snakuzzo** [reply]
E non puoi crearti la tua routine?

**Reply by KC-inDomus** [reply]
infatti, se non ti piace dire "accendi", creati una routine

**Reply by Larry-78** [reply]
Ciao, dopo aver esposto lo script verso Alexa, lo troverai tra gli scenari. Poi ti crei una routine e scegli "Quando questo accade" poi "Voce" a questo punto è possibile personalizzare la frase d'innesco. Dopo si sceglie di andare avanti per selezionare un'azione. Ovviamente la scelta sarà "Casa intelligente" poi "Controlla scenario" dove troverai il tuo script. Lascia lo stato su on se vuoi che Alexa lo accenda oppure seleziona off se vuoi che Alexa lo spenga.

**Reply by Dregi56** [reply]
Ok. grazie. strano che sia possibile la configurazione di routine da app su smartphone, ma non da sito web... mah!

**Reply by Larry-78** [reply]
Di nulla, la gestione da app è piu completa.
Solo a titolo informativo, creare una routine per far eseguire uno script non è l'unica soluzione, ci sarebbe anche l'utilizzo delle alexa actions, in pratica si da la possibilità ai dispositivi Alexa di fare delle interrogazioni e attende la nostra risposta, di conseguenza ci sarà l'action. Per esempio Alexa potrebbe dire: "Mi risulta che la luce del corridoio è rimasta accesa, vuoi spegnerla?" Alla nostra risposta "Si" lei spegne la luce. Questo si può fare con qualsiasi trigger e relativa action.
Se vuoi approfondire questo è il link.
https://github.com/keatontaylor/alexa-actions/wiki
